File: Utilities/ApiRoutes.cs
using System;
namespace Utilities
{
    public class ApiRoutes
    {
        public static class Base
        {
            /// <summary>
            /// Local
            /// </summary>
            public static readonly string BaseUrl = "http://localhost:52416/";

            //Android
            //public static readonly string BaseUrl = "http://localhost:52416/";

            /// <summary>
            /// Live
            /// </summary>
            //public static readonly string BaseUrl = "http://api.lycmcs.com/";
        }

        public static class Identity
        {
            private static readonly string BaseUrl = "api/Identity/";

            public static readonly string AuthenticateUser = $"{BaseUrl}AuthenticateUser";
            public static readonly string GetRoles = $"{BaseUrl}GetRoles";
            public static readonly string GetAllUsers = $"{BaseUrl}GetAllUsers";
            public static readonly string GetAllCustomers = $"{BaseUrl}GetAllCustomers";
            public static readonly string AddNewRole = $"{BaseUrl}AddNewRole";
            public static readonly string RegisterUser = $"{BaseUrl}RegisterUser";
            public static readonly string DeleteRole = $"{BaseUrl}DeleteRole";
            public static readonly string DeleteStaff = $"{BaseUrl}DeleteStaff";
            public static readonly string UpdateRole = $"{BaseUrl}UpdateRole";
            public static readonly string UpdateUser = $"{BaseUrl}UpdateUser";
            public static readonly string GetAllCustomersByBranchId = $"{BaseUrl}GetAllCustomersByBranchId";
        }

        public static class Customer
        {
            private static readonly string BaseUrl = "api/Customer/";

            public static readonly string AuthenticateUser = $"{BaseUrl}AddNewCustomer";
            public static readonly string GetAllCustomers = $"{BaseUrl}GetAllCustomers";
            public static readonly string GetAllCustomersByBranchId = $"{BaseUrl}GetAllCustomersByBranchId";

        }
    }
}

File: LYCPortal/Client/src/app/Shared/ApiRoutes/ApiRoutes.ts
export var apiRoutes = {
  BaseUrl: 'http://localhost:52416/',
  //BaseUrl: 'http://api.lycmcs.com/',
  Home: {
    GetUserByToken: 'api/Home/GetUserByToken'
  },
  Identity: {
    AuthenticateUser: 'api/IdentityResponsibility/AuthenticateUser',
    GetRoles: 'api/IdentityResponsibility/GetRoles',
    GetAllUsers: 'api/IdentityResponsibility/GetAllUsers',
    AddNewRole: 'api/IdentityResponsibility/AddNewRole',
    RegisterUser: 'api/IdentityResponsibility/RegisterUser',
    DeleteRole: 'api/IdentityResponsibility/DeleteRole',
    DeleteStaff: 'api/IdentityResponsibility/DeleteStaff',
    UpdateRole: 'api/IdentityResponsibility/UpdateRole',
    UpdateUser: 'api/IdentityResponsibility/UpdateUser'
  },

  Customer: {
    GetAllCustomers: 'api/IdentityResponsibility/GetAllCustomers',
    DeleteCustomer: 'api/IdentityResponsibility/DeleteCustomer',
    AddNewCustomer: 'api/IdentityResponsibility/AddNewCustomer',
    UpdateCustomer: 'api/IdentityResponsibility/UpdateCustomer'
  },
  Configuration: {
    Department: {
      GetDepartments: 'api/Department/GetDepartments',
      AddNewDepartment: 'api/Department/AddNewDepartment',
      UpdateDepartment: 'api/Department/UpdateDepartment',
      DeleteDepartment: 'api/Department/DeleteDepartment'
    },
    Branch: {
      AddNewBranch: 'api/Branch/AddNewBranch',
      GetAllBranches: 'api/Branch/GetBranches',
      DeleteBranch: 'api/Branch/DeleteBranch',
      UpdateBranch: 'api/Branch/UpdateBranch',
      GetBrancheByRegistrationNumber: 'api/Branch/GetBranchByRegistrationNumber',
      GetBranchFacilitiesByBranchId: 'api/Branch/GetBranchFacilitiesByBranchId'
    },
    Room: {
      GetRooms: 'api/Room/GetRooms',
      AddNewRoom: 'api/Room/AddNewRoom',
      UpdateRoom: 'api/Room/UpdateRoom',
      DeleteRoom: 'api/Room/DeleteRoom',
    },
    Product: {
      AddNewProduct: 'api/Product/AddNewProduct',
      GetProducts: 'api/Product/GetProducts',
      GetProductImage: 'api/Product/GetProductImage',
      DeleteProduct: 'api/Product/DeleteProduct',
      UpdateProduct: 'api/Product/UpdateProduct'
    },
    Package: {
      AddNewPackage: 'api/Package/AddNewPackage',
      GetPackages: 'api/Package/GetPackages',
      DeletePackage: 'api/Package/DeletePackage',
      UpdatePackage: 'api/Package/UpdatePackage'
    },
    Promotion: {
      AddNewPromotion: 'api/Promotion/AddNewPromotion',
      GetPromotions: 'api/Promotion/GetPromotions',
      DeletePromotion: 'api/Promotion/DeletePromotion',
      UpdatePromotion: 'api/Promotion/UpdatePromotion'
    },
    Service: {
      AddNewService: 'api/Service/AddNewService',
      GetServices: 'api/Service/GetServices',
      DeleteService: 'api/Service/DeleteService',
      UpdateService: 'api/Service/UpdateService'
    }
  },
  Finance: {
    GetFinances: 'api/Finance/GetFinances',
    AddNewFinance: 'api/Finance/AddNewFinance',
    DeleteFinance: 'api/Finance/DeleteFinance',
  }
}
